mula/tests/integration/test_scheduler.py

```diff
@@ -204,3 +208,73 @@ def test_enable_scheduler(self):
 
         # Stop the scheduler
         self.scheduler.stop()
+
+    # FIXME: deadline calculation is done in thread, so it's hard to test
+    def test_signal_handler_task(self):
```

Comment:
I remember us having some discussion about having deadlines and how/when to set them. But I remember an argument from my side where I mentioned that tasks can depend on a schedule, but if we update schedule deadlines through a signal when the tasks finish, we also create a dependency of the schedule on the running tasks, which sounded a bit tricky to me. Perhaps we can infer more at query time by looking at the most recent task?

Reply:
Yeah that would be an elegant solution, I'll look into it

Reply:
Yes so it boils down dropping the `deadline_at` field by calculating whether we've passed the deadline when we are going to reschedule jobs. I believe we could do that by calculating it the same way: using the `schedule` field and the `ended_at` of the last task

Reply:
Perhaps another argument:

If we schedule dns-scan to run on Mondays and nmap to run on Tuesdays, we might (?) see that nmap will already run on Monday due to it finding the ip adress. Then we run nmap anyways on Tuesdays using a `deadline_at`, because a lot can happend between now and the deadline that perhaps makes running the task at the deadline not necessary. The query-time approach of calculating whether we should run the task would handle this (if we make sure to match the schedule and tasks on task-hashes in that case?).